crawler/bitbucket.go

```diff
@@ -32,29 +37,81 @@ type response struct {
 
 // GetRepositories retrieves the list of all repository from an hosting.
 func (host Bitbucket) GetRepositories(repositories chan Repository) error {
-	var nextPage = host.URL
+	var sourceURL = host.URL
+
+	// Redis connection.
+	redisClient, err := redisClientFactory("redis:6379")
```

Comment:
we should invert the logic here, the Bitbucket implementation doesn't have to knows nothing about the fail management.

Reply:
Ok. I moved redis logic outside GetRepositories(). Now it saves the sourceURL if and only if it failed to retrieve the repositories list.